src/main/java/io/quarkus/fs/util/ZipUtils.java

```diff
@@ -183,29 +180,42 @@ public static FileSystem newFileSystem(URI uri, Map<String, Object> env) throws
             return FileSystemProviders.ZIP_PROVIDER.newFileSystem(uri, env);
         } catch (FileSystemAlreadyExistsException e) {
             throw new IOException("fs already exists " + uri, e);
-        } catch (IOException | ZipError ioe) {
-            // TODO: (at a later date) Get rid of the ZipError catching (and instead only catch IOException)
-            //  since it's a JDK bug which threw the undeclared ZipError instead of an IOException.
-            //  Java 9 fixes it https://bugs.openjdk.java.net/browse/JDK-8062754
-
+        } catch (IOException ioe) {
             // include the URI for which the filesystem creation failed
             throw new IOException("Failed to create a new filesystem for " + uri, ioe);
         }
     }
 
     /**
-     * This call is thread safe, a new FS is created for each invocation.
+     * Constructs a new FileSystem to access the contents of a zip as a file system.
      *
      * @param path The zip file.
      * @return A new FileSystem instance
      * @throws IOException in case of a failure
      */
     public static FileSystem newFileSystem(Path path) throws IOException {
+        return newFileSystem(path, Collections.emptyMap());
+    }
+
+    /**
+     * Constructs a new FileSystem to access the contents of a zip as a file system.
+     *
+     * @param path The zip file.
+     * @param env Property map to configure the constructed FileSystem.
+     * @return A new FileSystem instance
+     * @throws IOException in case of a failure
+     */
+    private static FileSystem newFileSystem(Path path, Map<String, Object> env) throws IOException {
+        if (env == null || env.isEmpty()) {
+            env = DEFAULT_OWNER_ENV;
+        } else {
+            env = new HashMap<>(env);
+            env.putAll(DEFAULT_OWNER_ENV);
```

Comment:
In case the passed in `env` already contains the `defaultOwner` and/or the `defaultGroup`, shouldn't this be reversed to
````
            final Map<String, Object> tmp = new HashMap<>(DEFAULT_OWNER_ENV);
            tmp.putAll(env);
            env = tmp;
````
?

Reply:
Good catch, thank you!